### PR TITLE
Core: Fix support for main.ts/preview.ts files

### PR DESCRIPTION
**Pull request by yannbf**
Issue: #11763

This is basically the same as #10099, for in depth information please check the description of that PR.

## What I did

TLDR: @babel/register is a dependency of a few libs in the monorepo, and that's why the examples using `main.ts` work. As soon as a user installs storybook in their projects, they don't have that dependency, which is required for server to be able to resolve ts files. 

## How to test

1 - start a new storybook project with storybook 6 or just use `sb upgrade` in an existing project
2 - run `yarn storybook` and see it working fine.
2 - rename main.js to main.ts
3 - run `yarn storybook` and see it failing.
4 - install `yarn add @babel/register -D`
5 - run `yarn storybook` and see it working fine again.

### CODE REVIEW

**Review by shilman — APPROVED** [review]
@yannbf LGTM! 🙏

**Review by jimmyandrade — COMMENTED** [review]
LGTM
